### PR TITLE
Support for qvm-template (qubes.Template{Search,Download} calls)

### PR DESCRIPTION
**Pull request by WillyPillow**
Discussions: <https://groups.google.com/forum/#!topic/qubes-devel/2XaMP4Us3kg>

Design doc: <https://gist.github.com/WillyPillow/b8a643ddbd9235a97bc187e6e44b16e4>

Major changes:

- New `qubes.Template{Search,Download}` qrexec calls for interaction with repos.

- [x] Deal with packaging.

### CODE REVIEW

File: qubes-rpc/qvm-template-repo-query [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+cat > "$repodir/template.repo"
+
+OPTS+=("--setopt=reposdir=${repodir}")
+OPTS+=("--quiet")
+
+RET=0
+
+if [ "$1" = "query" ]; then
+    # shellcheck disable=SC2068
+    dnf repoquery ${OPTS[@]} --qf='%{name}:%{epoch}:%{version}:%{release}:%{reponame}:%{downloadsize}:%{summary}' "$SPEC"
```
Comment (marmarek) [inline qubes-rpc/qvm-template-repo-query:33]: Shellcheck was right, `${OPTS[@]}` should be quoted.

File: qubes-rpc/qvm-template-repo-query [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    RET="$?"
+elif [ "$1" = "download" ]; then
+    # shellcheck disable=SC2068
+    curl -L "$(dnf download ${OPTS[@]} --url "$SPEC" | shuf -n 1)" -o -
```
Comment (marmarek) [inline qubes-rpc/qvm-template-repo-query:37]: And here too.

File: qubes-rpc/qvm-template-repo-query [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+SPEC=
+while IFS= read -r line; do
+    if [ "$line" = "---" ]; then
+        break
+    fi
+    case "$line" in
+        --enablerepo=*|\
+        --disablerepo=*|\
+        --repoid=*|\
+        --repofrompath=*|\
```
Comment (marmarek) [inline qubes-rpc/qvm-template-repo-query:13]: This one is not mentioned in the design doc, and also I don't think it's necessary.

Comment (WillyPillow) [reply qubes-rpc/qvm-template-repo-query]: I think I was considering whether to add it, also decided it wasn't necessary, but forgot to remove the line. Done.

**Review by marmarek — APPROVED** [review]

**Comment by marmarek** [comment]
The CI failures are unrelated, but we need to fix them before merging anyway. This will be done in #244 (nothing to worry about for you).

**Pull request merged**